MapboxCoreNavigation/RouteController.swift

```diff
@@ -832,8 +832,10 @@ extension RouteController: CLLocationManagerDelegate {
         if let accessToken = routeProgress.route.accessToken, let apiEndpoint = routeProgress.route.apiEndpoint, let host = apiEndpoint.host {
             directions = Directions(accessToken: accessToken, host: host)
         }
+        
+        let opts = NavigationRouteOptions(waypoints: options.waypoints, profileIdentifier: options.profileIdentifier)
```

Comment:
I'm not a fan of this change. I think if a developer uses map matching, they should actually never use our reroute logic as this would strip all options they have previously applied.